src/cromshell/cost/command.py

```diff
@@ -0,0 +1,302 @@
+                FROM `{bq_database}` as billing, UNNEST(labels) as wfid, UNNEST(labels) as task
+                WHERE cost > 0
+                AND task.key LIKE "wdl-task-name"
+                AND wfid.key LIKE "cromwell-workflow-id"
+                AND wfid.value like "%{workflow_id}"
+                AND partition_time BETWEEN "{start_date}" AND "{end_date}"
+                GROUP BY 1,2,3
+                ORDER BY 4 DESC
+                LIMIT 100
+            """
+        )
+    else:
+        query_job = client.query(
+            f"""
+                SELECT sum(cost) as cost
+                FROM `{bq_database}`, UNNEST(labels)
+                WHERE value = "cromwell-{workflow_id}" AND partition_time BETWEEN "{start_date}" AND "{end_date}"
+                LIMIT 100
+            """
+        )
+
+    query_results = query_job.result()
+
+    if query_results.total_rows == 0:
+        LOGGER.error("Could not retrieve cost - no cost entries found.")
+        raise ValueError("Could not retrieve cost - no cost entries found.")
+    else:
+        return query_results
+
+
+def minimum_time_passed_since_workflow_completion(
+    end_time: str, min_hours: int = 24
+) -> (bool, datetime):
+    """
+    Make sure 24 hours have passed between job finish time and executing this command
+    (finished time minus current time).
+
+    :param end_time: workflow completion time obtained from cromwell job metadata
+    :param min_hours:
+    :return:
+    """
+
+    delta = datetime.now(timezone.utc) - datetime.strptime(
+        end_time, "%Y-%m-%dT%H:%M:%S.%f%z"
+    )
+
+    mindelta = timedelta(hours=min_hours)
+
+    return (True, delta) if delta > mindelta else (False, delta)
+
+
+def check_cost_table_is_configured(config_options: dict) -> None:
+    if "bq_cost_table" not in config_options:
+        raise KeyError('Cromshell config file is missing "bq_cost_table" Key')
+
+    LOGGER.info("Using cost table: %s", config_options["bq_cost_table"])
+
+
+def get_metadata(config: object) -> dict:
+    import cromshell.metadata.command as metadata_command
+    from cromshell.utilities import http_utils
+
+    formatted_metadata_parameter = metadata_command.format_metadata_params(
+        list_of_keys=config.METADATA_KEYS_TO_OMIT,
+        exclude_keys=True,
+        expand_subworkflows=False,
+    )
+
+    return metadata_command.get_workflow_metadata(
+        meta_params=formatted_metadata_parameter,
+        api_workflow_id=config.cromwell_api_workflow_id,
+        timeout=config.requests_connect_timeout,
+        verify_certs=config.requests_verify_certs,
+        headers=http_utils.generate_headers(config),
+    )
+
+
+def get_submission_start_end_time(workflow_metadata: dict) -> (str, str):
+
+    events = workflow_metadata['workflowProcessingEvents']
+    try:
+        end_time = [i.get("timestamp") for i in events if i.get("description") == "Finished"][0]
```

Comment:
Yes ([source](https://github.com/broadinstitute/cromwell/blob/32d5d0cbf07e46f56d3d070f457eaff0138478d5/centaur/src/main/scala/centaur/CromwellTracker.scala#L22)), if the server was restarted. So with this code it will use the earliest start/end time. Would a warning or info log be useful here?